ansible_runner/runner.py

```diff
@@ -218,7 +218,7 @@ def _decode(x):
             if password is not None:
                 child.sendline(password)
                 self.last_stdout_update = time.time()
-            if self.cancel_callback:
+            if (not self.canceled) and self.cancel_callback:
```

Comment:
I think my concern here is... what if you need repeated calls to cancel?

Reply:
This doesn't affect how many times `Runner.handle_termination` is called. Do you mean to ask if a plugin needs multiple calls to `cancel_callback`?

Reply:
I pushed a commit to address this which has the test refactor I mentioned as a possibility. I'm worried that we're going too far being overly-conservative, to the point of protecting bad code practices.